security/iam/aws/open_iam_policy.pt

```diff
@@ -0,0 +1,137 @@
+  region 'us-east-1'
+  access_key cred('AWS_ACCESS_KEY_ID')
+  secret_key cred('AWS_SECRET_ACCESS_KEY')
+end
+
+
+# List of policies , extract useful info 
+datasource "ds_iam_policies" do
+  # This request is not paginated
+  request do
+    auth $auth_iam
+    host "iam.amazonaws.com"
+    path "/"
+    header "User-Agent", "RS Policies"
+    query "Version", "2010-05-08"
+    query "Action", "ListPolicies"
+  end
+  result do
+    encoding "xml"
+    collect xpath(response, "//ListPoliciesResponse/ListPoliciesResult/Policies/member", "array") do
+     field "name", xpath(col_item,"PolicyName")
+     field "id", xpath(col_item,"PolicyId")
+     field "arn", xpath(col_item,"Arn")
+     field "version_id", xpath(col_item,"DefaultVersionId")
+    end
+ end
+end
+
+datasource "ds_iam_policy_versions" do
+  iterate $ds_iam_policies
+  request do
+    auth $auth_iam
+    host "iam.amazonaws.com"
+    path "/"
+    header "User-Agent", "RS Policies"
+    query "Version", "2010-05-08"
+    query "Action", "GetPolicyVersion"
+    query "VersionId", val(iter_item, "version_id")
+    query "PolicyArn", val(iter_item, "arn")
+  end
+  result do
+    encoding "xml"
+    collect xpath(response, "//GetPolicyVersionResponse/GetPolicyVersionResult/PolicyVersion", "array") do
+     field "document", xpath(col_item,"Document")
+    end
+ end
+end
+
+
+
+
+#iterate over policies , undecode the document data for inspection.
+datasource "ds_readable_document" do
+  run_script $decode_and_correlate, $ds_iam_policy_versions, $ds_iam_policies
+end
+
+script "decode_and_correlate", type: "javascript" do
+  parameters "ds_iam_policy_versions", "ds_iam_policies"
+  result "name_and_policies"
+  
+  code <<-EOS
+  var name_and_policies=[]
+    _.each(ds_iam_policy_versions, function(policy_doc,index) {
+      var decoded_policy=JSON.parse(decodeURIComponent(policy_doc))
+
+      
+
+      name_and_policies.push({
+        name: ds_iam_policies[index].ListPoliciesResponse.ListPoliciesResult.Policies.PolicyName,
+        arn:  ds_iam_policies[index].ListPoliciesResponse.ListPoliciesResult.Policies.Arn,
+        id:   ds_iam_policies[index].ListPoliciesResponse.ListPoliciesResult.Policies.PolicyId,
+        action: _.first(decode_policy).Action,
+        effect: _.first(decode_policy).Effect,
+        resouce: _.first(decode_policy).Resource
+      })
+
+    });
+
+EOS
+end
+
+#allow access to everything
+# {
+#   "Version": "2012-10-17",
+#   "Statement": [
+#     {
+#       "Sid": "Stmt1528200363631",
+#       "Action": "*",
+#       "Effect": "Allow",
+#       "Resource": "*"
+#     }
+#   ]
+# }
+
+
+
+escalation "alert_aws" do
+   email $escalate_to
+end
+
+policy "iam_policies" do
+  validate_each $ds_readable_document do
+    summary_template "IAM Policies Found in AWS"
+    detail_template <<-EOS
+# Policies Found in AWS
+| Policy Name |  
+| ----------- |  
+{{ range data -}}
+| {{.name}} |  
+{{ end -}}
+EOS
+    escalate $alert_aws
+    check eq(1,0)
```

Comment:
always fail?  maybe check should only fail if there is data.  also what about showing ARN or other details to help the user.